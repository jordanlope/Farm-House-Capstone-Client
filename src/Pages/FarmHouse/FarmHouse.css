.FarmHouse section {
    text-align: center;
}