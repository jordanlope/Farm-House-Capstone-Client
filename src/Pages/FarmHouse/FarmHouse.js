import React, { Component } from 'react';
import './FarmHouse.css';
import FarmHousesService from '../../services/farmHouses-api-service'
import FarmHouseContext from '../../contexts/FarmHouseContext'

export default class FarmHouse extends Component {
  static contextType = FarmHouseContext

  componentDidMount() {
    this.context.clearError()
    FarmHousesService.getFarmHouse(this.props.match?.params.homeId)
      .then(this.context.setFarmHouse)
      .catch(this.context.setError)
  }

  render() {
    const { farmHouse } = this.context ? this.context : {}
    const { realtor } = this.context

    console.log('This is current realtor', realtor)
    return (
      <section className="FarmHouse">
        <h2>Farm House Details</h2>
        <p>{"Address: " + farmHouse.address}</p>
        <p>{"Description: " + farmHouse.description}</p>
        <p>{"Price: $" + farmHouse.price}</p>
        <h2>Realtor Contact Info</h2>
        <p>{"Name: " + realtor.full_name}</p>
        <p>{"Email: " + realtor.email}</p>
        <p>{"Number: " + realtor.number}</p>
        <p>{"Description: " + realtor.description}</p>
      </section>
    );
  }
}