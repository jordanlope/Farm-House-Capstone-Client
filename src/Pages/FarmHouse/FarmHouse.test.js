import React from 'react';
import ReactDOM from 'react-dom';
import FarmHouse from './FarmHouse';
import renderer from 'react-test-renderer';

describe('FarmHouse component', () => {
    // Smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<FarmHouse />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    // Snapshot test
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<FarmHouse />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})