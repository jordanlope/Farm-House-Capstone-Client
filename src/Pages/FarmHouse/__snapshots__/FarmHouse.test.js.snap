// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FarmHouse component renders the UI as expected 1`] = `
<section
  className="FarmHouse"
>
  <h2>
    Farm House Details:
  </h2>
  <p>
    Address: undefined
  </p>
  <p>
    Description: undefined
  </p>
  <p>
    Price: $undefined
  </p>
  <h2>
    Realtor Contact Info:
  </h2>
  <p>
    Name: undefined
  </p>
  <p>
    Email: undefined
  </p>
  <p>
    Number: undefined
  </p>
  <p>
    Description: undefined
  </p>
</section>
`;
